File: app/src/main/java/org/stepic/droid/util/ProgressUtil.java
package org.stepic.droid.util;

import org.stepic.droid.model.IProgressable;

import java.util.ArrayList;
import java.util.List;

public class ProgressUtil {
    public static String[] getAllProgresses(List<? extends IProgressable> objects) {

        List<String> progressesId = new ArrayList<>();
        for (IProgressable item : objects) {
            progressesId.add(item.getProgressId());
        }
        return progressesId.toArray(new String[progressesId.size()]);
    }
}

File: app/src/main/java/org/stepic/droid/util/ProgressUtil.kt
package org.stepic.droid.util

import org.stepik.android.model.Progressable

object ProgressUtil {
    fun getProgresses(objects: Iterable<Progressable>?): Array<String> {
        return objects
                ?.mapNotNull { it.progress }
                ?.toTypedArray()
                ?: emptyArray()
    }
}

@JvmName("progressableIterable_getProgresses")
fun Iterable<Progressable>.getProgresses(): Array<String> =
    ProgressUtil.getProgresses(this)
